src/lib.rs

```diff
@@ -27,39 +30,99 @@ pub mod user;
+    Other(String, StatusCode),
 }
 
+impl Error {
+    fn info_string(&self) -> &str {
+        match self {
+            Self::BadAuth => "You are not authenticated.",
+            Self::InvalidName => "Invalid name.",
+            Self::Banned => "You are banned from that hub.",
+            Self::ChannelNotFound => "Channel not found.",
+            Self::Deserialize => "Server was unable to deserialize the data. Try again later.",
+            Self::Directory => {
+                "Server is missing a directory and could not create it. Try again later."
+            }
+            Self::GroupNotFound => "Group not found.",
+            Self::HubNotFound => "Hub not found.",
+            Self::MemberNotFound => "Hub member not found.",
+            Self::Muted => "You are muted.",
+            Self::NoPermission => "You do not have permission to do that.",
+            Self::NotInHub => "You are not in that hub.",
+            Self::ReadFile => "Server was unable to read the data. Try again later.",
+            Self::Serialize => "Server was unable to serialize the data. Try again later.",
+            Self::UserNotFound => "User not found.",
+            Self::WriteFile => "Server was unable to store the data. Try again later.",
+            Self::DeleteFailed => "Server was unable to delete the data.",
+            Self::UnexpectedServerArg => "Something strange happened...",
+            Self::MessageTooBig => "Message too big.",
+            Self::InvalidMessage => "Messages must be sent as UTF-8 strings.",
+            Self::Other(message, _) => message,
+        }
+    }
+
+    fn http_status_code(&self) -> StatusCode {
+        match self {
+            Self::BadAuth => StatusCode::UNAUTHORIZED,
+            Self::InvalidName => StatusCode::BAD_REQUEST,
+            Self::Banned => StatusCode::FORBIDDEN,
+            Self::ChannelNotFound => StatusCode::NOT_FOUND,
+            Self::Deserialize => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::Directory => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::GroupNotFound => StatusCode::NOT_FOUND,
+            Self::HubNotFound => StatusCode::NOT_FOUND,
+            Self::MemberNotFound => StatusCode::NOT_FOUND,
+            Self::Muted => StatusCode::FORBIDDEN,
+            Self::NoPermission => StatusCode::FORBIDDEN,
+            Self::NotInHub => StatusCode::NOT_FOUND,
+            Self::ReadFile => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::Serialize => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::UserNotFound => StatusCode::NOT_FOUND,
+            Self::WriteFile => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::DeleteFailed => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::UnexpectedServerArg => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::MessageTooBig => StatusCode::BAD_REQUEST,
+            Self::InvalidMessage => StatusCode::BAD_REQUEST,
+            Self::Other(_, code) => code.clone(),
+        }
+    }
+}
+
```

Comment:
consider using `thiserror` which provides macros that generates implementations for you.

Reply:
I switched to this error